### DISCUSSION TITLE
paddleocr 在华为910b的aarch架构的npu上运行很慢且只跑了一个卡

### DISCUSSION THREAD

**Discussion by danyXu** [PaddlePaddle/PaddleOCR#12530] [Q&A]
[参考此文档]( https://github.com/PaddlePaddle/PaddleCustomDevice/blob/develop/backends/npu/README_cn.md)  在华为910b的arrch架构机器上成功安装了paddleocr 

paddle-custom-npu 0.0.0
paddle2onnx 1.0.5
paddleclas 2.5.2
paddleocr 2.7.3
paddlepaddle 3.0.0.dev20240527 

但简单使用paddleocr命令，带上参数use_npu=true,发现两个问题：

1. 我机器上有8张卡，运行只使用了一张，也没有参数来设置，如下图：
执行参数：

`Namespace(help='==SUPPRESS==', use_gpu=False, use_xpu=False, use_npu=True, ir_optim=True, use_tensorrt=False, min_subgraph_size=15, precision='fp32', gpu_mem=500, gpu_id=0, image_dir='imgs/123fail_0.png', page_num=0, det_algorithm='DB', det_model_dir='/root/.paddleocr/whl/det/ch/ch_PP-OCRv4_det_infer', det_limit_side_len=960, det_limit_type='max', det_box_type='quad', det_db_thresh=0.3, det_db_box_thresh=0.6, det_db_unclip_ratio=1.5, max_batch_size=10, use_dilation=False, det_db_score_mode='fast', det_east_score_thresh=0.8, det_east_cover_thresh=0.1, det_east_nms_thresh=0.2, det_sast_score_thresh=0.5, det_sast_nms_thresh=0.2, det_pse_thresh=0, det_pse_box_thresh=0.85, det_pse_min_area=16, det_pse_scale=1, scales=[8, 16, 32], alpha=1.0, beta=1.0, fourier_degree=5, rec_algorithm='SVTR_LCNet', rec_model_dir='/root/.paddleocr/whl/rec/ch/ch_PP-OCRv4_rec_infer', rec_image_inverse=True, rec_image_shape='3, 48, 320', rec_batch_num=6, max_text_length=25, rec_char_dict_path='/usr/local/lib/python3.10/dist-packages/paddleocr/ppocr/utils/ppocr_keys_v1.txt', use_space_char=True, vis_font_path='./doc/fonts/simfang.ttf', drop_score=0.5, e2e_algorithm='PGNet', e2e_model_dir=None, e2e_limit_side_len=768, e2e_limit_type='max', e2e_pgnet_score_thresh=0.5, e2e_char_dict_path='./ppocr/utils/ic15_dict.txt', e2e_pgnet_valid_set='totaltext', e2e_pgnet_mode='fast', use_angle_cls=False, cls_model_dir='/root/.paddleocr/whl/cls/ch_ppocr_mobile_v2.0_cls_infer', cls_image_shape='3, 48, 192', label_list=['0', '180'], cls_batch_num=6, cls_thresh=0.9, enable_mkldnn=False, cpu_threads=10, use_pdserving=False, warmup=False, sr_model_dir=None, sr_image_shape='3, 32, 128', sr_batch_num=1, draw_img_save_dir='./inference_results', save_crop_res=False, crop_res_save_dir='./output', use_mp=False, total_process_num=1, process_id=0, benchmark=False, save_log_path='./log_output/', show_log=True, use_onnx=False, output='./output', table_max_len=488, table_algorithm='TableAttn', table_model_dir=None, merge_no_span_structure=True, table_char_dict_path=None, layout_model_dir=None, layout_dict_path=None, layout_score_threshold=0.5, layout_nms_threshold=0.5, kie_algorithm='LayoutXLM', ser_model_dir=None, re_model_dir=None, use_visual_backbone=True, ser_dict_path='../train_data/XFUND/class_list_xfun.txt', ocr_order_method=None, mode='structure', image_orientation=False, layout=True, table=True, ocr=True, recovery=False, use_pdf2docx_api=False, invert=False, binarize=False, alphacolor=(255, 255, 255), lang='ch', det=True, rec=True, type='ocr', ocr_version='PP-OCRv4', structure_version='PP-StructureV2')`

![image](https://github.com/PaddlePaddle/PaddleOCR/assets/3701158/e2694df3-2617-44ba-a748-37d05d05ff60)

3. 跑的很慢，cpu只9秒，npu跑几分钟，一个简单图
运行参数：
- cpu运行
   
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/3701158/3bf661a2-5366-4c69-9f1e-6aa946eabfbc)

- npu运行
  
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/3701158/80abe187-e2b4-45c5-bc89-c2e544f299a7)

**Comment by SWHL** [comment]
好奇这个问题有解决吗？

**Comment by lizhaoliaooo** [comment]
后面问题有解决么
